### DISCUSSION TITLE
Deploy Next.js app

### DISCUSSION THREAD

**Discussion by LouisWinkler** [community/community#63669] [New to GitHub]
I'm having trouble deploying my Next.js application using GitHub and I'm hoping someone can help me out. I've created a new repository in GitHub and pushed my code to it, but I'm unsure of the best approach for hosting my app.

Can anyone advise on the following questions?

What is the recommended way to host a Next.js app using GitHub?
I think there is a way to deploy next.js app using repository in github.

Any help or advice would be greatly appreciated. 
Please help me. thanks.

**Comment by ChristianSoh72** [comment]
There are many free hosting sites.
Did you do googling about deploying?
Vercel would be great selection.

Any further question?

**Comment by nirav-gajera** [comment]
- GitHub Pages: GitHub Pages is a free service that allows you to host static websites on GitHub. This is a good option if your Next.js app is static, or if you only need to host a small number of pages.

- Vercel: Vercel is a hosting platform that is specifically designed for Next.js apps. Vercel offers a number of features that make it easy to deploy and scale Next.js apps, such as automatic deployments, automatic HTTPS, and support for serverless functions.

- Netlify: Netlify is another hosting platform that is specifically designed for Next.js apps. Netlify offers similar features to Vercel, such as automatic deployments, automatic HTTPS, and support for serverless functions.

**Comment by pugalkmc** [comment]
I have a working solution with some guidance

Recently , i was in search for free server deployment , render is the only option for free server deployments but render will make your server into sleep mode when there is no request made

So i found a solution , i hope it may helpful for you and save your money
There is a consept called serverless

Which you can use your row file path as a server
example:
/api/user
To activate this above route with serverless technique , you need to create api folder and inside a user file
Same like you will create nested folder for each nested route
This is all about serverless technique

Now we can use express js code directly to deploy using serverless technique , the change you need to make is the below
remove app.listen(PORT)
Add export default app

See the image
![commanded app listen()](https://github.com/community/community/assets/88258254/df819cb1-0f22-423d-94f8-683d3a067bea)
Here i use express , so commanded my server start line only , so you need to command your single line of code which starts your server

![add api folder](https://github.com/community/community/assets/88258254/715bcf99-8104-4607-8a39-6aa098a87c2f)
Here see the /api folder

Inside i have index.js file and imported app and exported again
this way , you don't need to change your express js code and you can now deploy your code on vercel for free

How to deploy express js server with serverless technique with vercel:
https://www.youtube.com/watch?v=B-T69_VP2Ls

Vercel Docs:
https://vercel.com/guides/using-express-with-vercel
